File: ANCIEN BACK/cible/domain/stock/services/consumables_service.py
from cible.domain.stock.interface.i_consumables_repository import IConsumablesRepository
from cible.domain.stock.models.consumables_glues_bean import ConsumablesGluesBean
from cible.domain.stock.models.other_consumables_bean import OtherConsumablesBean


def get_all_consumables_glues(
    repository: IConsumablesRepository,
) -> list[ConsumablesGluesBean]:
    return repository.get_all_consumables_glues()


def get_all_other_consumables(
    repository: IConsumablesRepository,
) -> list[OtherConsumablesBean]:
    return repository.get_all_other_consumables()


def delete_consumables_glues(
    uuid: str, repository: IConsumablesRepository
) -> list[ConsumablesGluesBean]:
    return repository.delete_consumables_glues(uuid)


def delete_other_consumables(
    uuid: str, repository: IConsumablesRepository
) -> list[OtherConsumablesBean]:
    return repository.delete_other_consumables(uuid)


def create_consumables_glues(
    repository: IConsumablesRepository, consumables_glues_bean: ConsumablesGluesBean
) -> list[ConsumablesGluesBean]:
    return repository.create_consumables_glues(consumables_glues_bean)


def create_other_consumables(
    repository: IConsumablesRepository, other_consumables_bean: OtherConsumablesBean
) -> list[OtherConsumablesBean]:
    return repository.create_other_consumables(other_consumables_bean)


def create_consumables_glues_from_excel_sheet(
    repository: IConsumablesRepository, consumables_glues_bean: ConsumablesGluesBean
) -> None:
    return repository.create_consumables_glues_from_excel_sheet(consumables_glues_bean)


def create_other_consumables_from_excel_sheet(
    repository: IConsumablesRepository, other_consumables_bean: OtherConsumablesBean
) -> None:
    return repository.create_other_consumables_from_excel_sheet(other_consumables_bean)


def add_additional_comment_to_consumables_glues(
    repository: IConsumablesRepository, comment: str, object_uuid: str
) -> None:
    return repository.add_additional_comment_to_consumables_glues(comment, object_uuid)


def add_additional_comment_to_other_consumables(
    repository: IConsumablesRepository, comment: str, object_uuid: str
) -> None:
    return repository.add_additional_comment_to_other_consumables(comment, object_uuid)

File: ANCIEN FRONT/services/stock/consumables.service.ts
import {fetcher} from "../utils/fetch-utils.service.ts";
import {BASE_API_CIBLE_HREF} from "../../constants.ts";
import ConsumablesGluesModel from "../../core/domain/stock/ConsumablesGluesModel.ts";
import OtherConsumablesModel from "../../core/domain/stock/OtherConsumablesModel.ts";

export function getAllConsumablesGlues(): Promise<ConsumablesGluesModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-consumables-glues`, requestOptions);
}

export function getAllOtherConsumables(): Promise<OtherConsumablesModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-other-consumables`, requestOptions);
}

export function deleteConsumablesGlues(uuid: string): Promise<ConsumablesGluesModel[]> {
    const requestOptions: RequestInit = {
        method: 'DELETE',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/consumables-glues/${uuid}`, requestOptions);
}

export function deleteOtherConsumables(uuid: string): Promise<OtherConsumablesModel[]> {
    const requestOptions: RequestInit = {
        method: 'DELETE',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/other-consumables/${uuid}`, requestOptions);
}

export function createConsumablesGlues(object : ConsumablesGluesModel): Promise<ConsumablesGluesModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/consumables-glues`, requestOptions);
}

export function createOtherConsumables(object: OtherConsumablesModel): Promise<OtherConsumablesModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/other-consumables`, requestOptions);
}
export function addAdditionalCommentToConsumablesGlues(comment: string, uuid: string): Promise<null> {
    const requestOptions: RequestInit = {
        method: 'PUT',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({comment, uuid})

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/consumables-glues`, requestOptions);
}
export function addAdditionalCommentToOtherConsumables(comment: string, uuid: string): Promise<null> {
    const requestOptions: RequestInit = {
        method: 'PUT',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({comment, uuid})

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/other-consumables`, requestOptions);
}